builtin/mainmenu/tab_online.lua

```diff
@@ -264,19 +265,49 @@ end
 
 --------------------------------------------------------------------------------
 
+local function parse_search_input(input)
+	-- Return nil if nothing to search for
+	if not input:find("%S") then
+		return nil
+	end
+
+	local query = {keywords = {}, mods = {}, players = {}}
+
+	-- Process quotation enclosed parts
+	input = input:gsub("([^ ]?)\"([^\"]*)\"([^ ]?)", function(before, match, after)
+		if #before == 0 and #after == 0 then -- Also have be separated by spaces
+			table.insert(query.keywords, match)
+			return " "
+		end
+		return before..'"'..match..'"'..after
+	end)
+
+	-- Separate by space characters and handle special prefixes
+	-- (words with special prefixes need an exact match and none of them can contain spaces)
+	for word in input:gmatch("%S+") do
+		local mod = word:match("^mod:(.*)")
+		table.insert(query.mods, mod)
```

Comment:
Why not lowercase `mod`, `player`, `game` directly here? That would be consistent with the `table.insert(query.keywords, word:lower())` below.

Reply:
Oh, I missed to remove the `word:lower()` you are mentioning.
The whole search string gets lowered before it gets parsed. This was also the case before this PR, so I readded it since you wrote that player, game ... should not be case sensitive.
(Before I made it case sensitive exactly such that you could distinguish between players/games ... with different capitalization, e.g. `exile`, `Exile`. But I think non case sensitivity is the way to go.)

Now I also moved `input = input:lower()` into the `parse_search_input` function to make it even more clear.